### ISSUE TITLE
Tests failing on a clean repository clone

### ISSUE THREAD

**Issue by nouritsu** [uutils/coreutils#5130]
Replicate -
- Run `git clone https://github.com/uutils/coreutils && cd coreutils && cargo test`
- Read message returned by `cargo test`

Host specs -
- Windows 11 64-bit Latest (as of 3rd August 2023)
- AMD Ryzen 5 3600x
- 16GB 3800MHz RAM
- Used PowerShell in VS Code

**Comment by tertsdiepraam** [comment]
What is the message from `cargo test`?

**Comment by nouritsu** [comment]
Attached the message from `cargo test`, used
```
cargo test > out.txt
```
[out.txt](https://github.com/uutils/coreutils/files/12249044/out.txt)

**Comment by nouritsu** [comment]
Note that stderr only reports compiles, module runs and test rerun message ; hence not included in the text file.
![image](https://github.com/uutils/coreutils/assets/113834791/30e105e2-5ffd-4d09-a03f-af01212101d1)

**Comment by tertsdiepraam** [comment]
Right, so `touch` is consistently 200 seconds off from the expected value?

**Comment by nouritsu** [comment]
I'm in India, which has the IST time zone. It is `UTC + 5:30`
Here is a screenshot from windows settings -
![image](https://github.com/uutils/coreutils/assets/113834791/7b08827a-785e-4707-b1b8-f118ba8f422a)

**Comment by tertsdiepraam** [comment]
Oh very interesting, I guess it might not be timezone-related then or maybe it is, but in some more complicated way 🤔

**Comment by nouritsu** [comment]
The problem may be with `chrono` getting the incorrect time for whatever reason. Would you like me to run some code that would allow us to rule `chrono` out?

**Comment by tertsdiepraam** [comment]
That would be great! I was thinkin the culprit could be the conversion from `chrono` to `filetime`. But yeah, let's rule `chrono` out first.

**Comment by nouritsu** [comment]
I'm not familiar with the crate but I ran this code -
```rs
use chrono::prelude::*;

fn main() {
    let local = Local::now();
    println!("{:?}", local);
}
```
Using `chrono v0.4.26`.

Output -
![image](https://github.com/uutils/coreutils/assets/113834791/0a39b050-fd51-4057-8ac3-1164119897fa)
This seems to be correct

**Comment by nouritsu** [comment]
A little reading later, I think I understand how to test if `chrono` to `filetime` is an issue.
Here is the code -
```rs
use chrono::prelude::*;
use filetime::FileTime;
use std::time::{SystemTime, UNIX_EPOCH};

fn main() {
    let ftime = datetime_to_filetime(&Local::now()).unix_seconds();
    let stime = SystemTime::now()
        .duration_since(UNIX_EPOCH)
        .unwrap()
        .as_secs();
    println!("{}\t{}", ftime, stime);
}

fn datetime_to_filetime<T: TimeZone>(dt: &DateTime<T>) -> FileTime {
    FileTime::from_unix_time(dt.timestamp(), dt.timestamp_subsec_nanos())
}
```

Prints the following -
```
1691056504      1691056504
```
Note that it is 15:25 3rd August 2023

**Comment by tertsdiepraam** [comment]
So that's correct? Just as a sanity check, could you run one of the failing test cases manually in the terminal and check whether the time is correct? It might be an issue with the tests, instead of an issue with `touch`.

**Comment by nouritsu** [comment]
Executing command -
```
coreutils pr --pages=3:5 --column=3 -a -n ".\tests\fixtures\pr\column.log" > pr.txt
```
Test - `test_pr::test_with_column_across_option`
Times seem to be wrong.
[pr.txt](https://github.com/uutils/coreutils/files/12249575/pr.txt)

**Comment by nouritsu** [comment]
Running the command again a few minutes later displayed the time as following (in the output) `Aug 03 13:18 2023`
Whereas it should be `Aug 03 15:44 2023`

**Comment by tertsdiepraam** [comment]
Strange, we don't do much in `pr`, just formatting the last modified time. Could you maybe try a small program that just gets a system time, converts it to `DateTime<Local>` and formats it (like `pr` does)?

Maybe the conversion from system time to chrono is wrong?

**Comment by nouritsu** [comment]
Yes,
```rs
use chrono::prelude::*;
use filetime::FileTime;
use std::time::SystemTime;

const DATE_TIME_FORMAT: &str = "%b %d %H:%M %Y";

fn main() {
    let time: DateTime<Local> = SystemTime::now().into();
    println!("{}", time.format(DATE_TIME_FORMAT));
}
```
outputs
```
Aug 03 18:12 2023
```
Which is correct

**Comment by nouritsu** [comment]
Any idea why this was failing? I'd love to test this again tomorrow and close this issue.

**Comment by tertsdiepraam** [comment]
I'm still not sure to be honest

**Comment by nouritsu** [comment]
This does not seem to be the case anymore. Closing the issue.
